### PR TITLE
Disable flaky test - testKillBasicFlowWithoutEndNode

### PR DESCRIPTION
**Pull request by jamiesjc**
Disable the flaky test for now to unblock the travis build.
See the detailed discussion in #1921

### CODE REVIEW

File: azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerYamlTest.java [hunk 1]
```diff
@@ -56,6 +57,7 @@ public void testBasicFlowWithoutEndNode() throws Exception {
     assertFlowStatus(flow, Status.SUCCEEDED);
   }
 
+  @Ignore
```
Comment (juhoautio) [inline azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerYamlTest.java]: Could include a comment, something like this:

There seems to be an actual race condition bug in the runtime code. See: issue #1921: Flaky test FlowRunnerTestYaml & issue #1311: Potential race condition between flowRunner thread and jetty killing thread

Comment (jamiesjc) [reply azkaban-exec-server/src/test/java/azkaban/execapp/FlowRunnerYamlTest.java]: Good suggestion! I will update it.

**Comment by HappyRay** [comment]
@jamiesjc 
What's the plan for this test next? Is there a ticket to track the remaining work?
If this test has no value, it should be removed.
If it is flaky because of test implementation, could you try fixing it? 
If the test failure is signaling us that there is a bug in the production code, you know what to do :)

**Comment by jamiesjc** [comment]
@HappyRay, see my comments in #1921:

> When killing happens, jobC was not even added to activeJobRunners even though it was already submitted. We've seen this race condition in killing flows before and it would take much effort to completely fix it. I would disable this test for now to unblock the travis build. With the new DAG engine design, we should not see this kind of issue again. 
> 
> ```
> 2018/09/06 14:02:37.544 +0000 INFO [FlowRunner-exec-116] [basic_flow] Submitting job 'jobC' to run.
>     2018/09/06 14:02:37.540 +0000 INFO [Test worker] [basic_flow] Kill has been called on flow 116
>     2018/09/06 14:02:37.544 +0000 INFO [Test worker] [basic_flow] Killing 0 jobs.
>     2018/09/06 14:02:37.545 +0000 INFO [JobRunner-jobC-116] [basic_flow] Created file appender for job jobC
>     2018/09/06 14:02:37.545 +0000 INFO [JobRunner-jobC-116] [basic_flow] Attached file appender for job jobC
>     2018/09/06 14:02:37.545 +0000 INFO [JobRunner-jobC-116] [basic_flow] Job Started: jobC
> ```

You might still remember the original race condition issue we discussed in #1311. I proposed a fix before but it could not completely resolve this issue so I aborted it. I think you might have some fix for that? Anyway, this would not be a problem once we change to the new DAG engine model. 
I can create a ticket to track this issue internally.

**Comment by HappyRay** [comment]
@jamiesjc 
Thanks for the explanation. I saw your comment in the test now. It is great that you are documenting why a test is disabled.

I saw that #1311 is still open. I don't remember the exact details now.  That references my previous fix #1310. Is that fix what you are referring to?

I am too busy on other projects to continue the work on the new DAG engine. There is still a lot of work to do. With the added complexity of conditional workflow, the integration work will take even longer. 

If you know a fix already for the existing code, I wouldn't wait for the DAG engine to replace this code.

**Comment by jamiesjc** [comment]
There were several potential race condition issues we discovered at that time.  #1310 addressed one of them. I think there is still one issue remaining as mentioned in the change description:
```
Scope of the fix:

This fix only addresses the synchronization issue between the killing Jetty thread and the jobrunner thread. It doesn't address the potential synchronization issues between the killing thread and the flowrunner thread e.g. There may be still some other race condition that causes a kill action to not kill a running job.
```
I don't remember all the details. But I can take a look at it and try to refresh my memory.

**Pull request merged**